File: easy/go/1732-find-the-highest-altitude.go
package main

func largestAltitude(gain []int) int {
	currentAltitude := 0
	maxAltitude := 0 

	for i := 0 ; i < len(gain); i++ {
		currentAltitude += gain[i]
		if currentAltitude > maxAltitude {
			maxAltitude = currentAltitude
		}
	}

	return maxAltitude
}

File: easy/javascript/1732-find-the-highest-altitude.js
/** O(n)
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function (gain) {
  let currentAltitude = 0;
  let maxAltitude = 0;

  for (let i = 0; i < gain.length; i++) {
    currentAltitude += gain[i];
    if (currentAltitude > maxAltitude) {
      maxAltitude = currentAltitude;
    }
  }

  return maxAltitude;
};
